modules/python/python-logmsg.c

```diff
@@ -131,21 +132,15 @@ _py_log_message_ass_subscript(PyObject *o, PyObject *key, PyObject *value)
 
   NVHandle handle = log_msg_get_value_handle(name);
 
-  if (value && is_py_obj_bytes_or_string_type(value))
-    {
-      const gchar *str;
-      py_bytes_or_string_to_string(value, &str);
-      log_msg_set_value(py_msg->msg, handle, str, -1);
-    }
-  else
-    {
-      PyErr_Format(PyExc_ValueError,
-                   "str or unicode object expected as log message values, got type %s (key %s). "
-                   "Earlier syslog-ng accepted any type, implicitly converting it to a string. "
-                   "With this version please convert it explicitly to a string using str()",
-                   value->ob_type->tp_name, name);
-      return -1;
-    }
+  if (!value)
+    return -1;
+
+  gchar *log_msg_value;
+  LogMessageValueType type;
+  if (!py_obj_to_log_msg_value(value, &log_msg_value, &type))
+    return -1;
+
+  log_msg_set_value_with_type(py_msg->msg, handle, log_msg_value, -1, type);
```

Comment:
this is leaking memory, as log_msg_value() is returned by py_obj_to_log_msg_value() as an allocated string.

could we perhaps use scratch buffers for this purpose?

Reply:
Oh yes, I missed the `g_free()` here.

I could convert `py_obj_to_log_msg_value()` to work on `GString *`s, but adding the missing `g_free()` should suffice as well, I guess.

`GString` with `ScratchBuffers` would improve the performance as well, so I would prefer that.

Reply:
Modified `py_obj_to_log_msg_value()` to append to a given `GString`.